suite-native/passphrase/src/useFeatureRequestsPassphrase.tsx

```diff
@@ -0,0 +1,68 @@
+import { useCallback, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import TrezorConnect from '@trezor/connect';
+import {
+    PassphraseStackParamList,
+    PassphraseStackRoutes,
+    RootStackParamList,
+    RootStackRoutes,
+    StackToStackCompositeNavigationProps,
+} from '@suite-native/navigation';
+import {
+    deviceActions,
+    selectDevice,
+    selectDeviceButtonRequestsCodes,
+} from '@suite-common/wallet-core';
+
+type NavigationProp = StackToStackCompositeNavigationProps<
+    PassphraseStackParamList,
+    PassphraseStackRoutes,
+    RootStackParamList
+>;
+
+// When device is in view only mode and is reconnected, passphrase is not requested after reconnecting
+// but when the app interacts with the device (e.g. requesting address / validating transaction, etc).
+// To make the passphrase working on this feature, just use this hook in the component that will interact with the device
+// and add the features button request code to the useEffect that redirects back to the flow
+// after successfull passphrase submission.
+export const useFeatureRequestsPassphrase = () => {
```

Comment:
Why is this not handled globally?

Reply:
It's different treatment to other `ui-request_passphrase` events so the idea was to keep it scoped together with the functionality that requires this. I'd rather have two simple hooks with single purpose than one overcomplicated hook with conditions that would be hard to read and understand. If it was all handled within one `ui-request_passphrase` handler, I'm not sure what would the condition be to decide whether to navigate to passphrase form for wallet "creation" or this UI.

Reply:
I'm afraid this is more complicated than global handler and it is easy to miss handling passhprase while adding new feature... 

The condition can be if the selected device is already discovered or new device.

Reply:
After mutual discussion, I moved the UI.REQUEST_PASSPHRASE listener to one global hook that will handle these redirects.
https://github.com/trezor/trezor-suite/pull/12677/commits/b6d6469c0d616015be50aefe21b6a1c44d101ac0